scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/policy/ram/RamCompute.java

```diff
@@ -0,0 +1,39 @@
+package org.ow2.proactive.scheduler.policy.ram;
+
+import java.io.Serializable;
+import java.lang.management.ManagementFactory;
+
+import org.objectweb.proactive.core.runtime.LocalNode;
+import org.objectweb.proactive.core.runtime.ProActiveRuntimeImpl;
+
+import com.sun.management.OperatingSystemMXBean;
+
+
+public class RamCompute implements Serializable {
+
+    private static final long GIGABYTE = (1024L * 1024L * 1024L);
+
+    public RamCompute() {
+    }
+
+    public long getAvailableRAM() {
+
+        OperatingSystemMXBean bean = (OperatingSystemMXBean) ManagementFactory.getOperatingSystemMXBean();
+
+        long totRAMAvailable = bean.getTotalPhysicalMemorySize() / GIGABYTE;
+
+        int totalRAMUsed = 0;
+        for (LocalNode localNode : ProActiveRuntimeImpl.getProActiveRuntime().getLocalNodes()) {
+            if (localNode.getProperty(RamSchedulingPolicy.RAM_VARIABLE_NAME) != null) {
+                totalRAMUsed += Integer
+                        .parseInt(localNode.getProperty(RamSchedulingPolicy.RAM_VARIABLE_NAME));
+            }
+        }
+
+        long ramInBytes = totRAMAvailable - totalRAMUsed;
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Immediately return this expression instead of assigning it to the temporary variable "ramInBytes". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://master.ci.activeeon.com:9000/coding_rules#rule_key=squid%3AS1488)